src/components/smallComponents/SnackbarWithButton.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "state/store";
+import { setSnackbarClose } from "state/actions/snackbarActions";
+
+import { Snackbar } from "@material-ui/core";
+import { Alert, Color } from "@material-ui/lab";
+
+const SnackbarWithButton: React.FC<{}> = () => {
+  const dispatch = useDispatch();
+  const { open, message, severity } = useSelector(
+    (state: RootState) => state.snackbarReducer
+  );
```

Comment:
I think handling the open state from the snackbarReducer is a bit too much. Basically, it can be kept in your own component state.
Then you could use 'useEffect' on the 'message' and 'severity' value to set it to true.
And to finish set it to false in your 'handleClose' function, instead of your 'dispatch' call.